burn-core/src/optim/sgd.rs

```diff
@@ -124,6 +126,13 @@ mod tests {
         assert!(record.is_empty());
     }
 
+    #[test]
+    fn can_attach_gradient_clipper() {
+        let optim = sgd_with_all().with_gradient_clipper(GradientClipper::ClipByValue(0.5));
+        let record = optim.to_record();
+        assert!(record.is_empty());
```

Comment:
Does the test fail if you remove `with_gradient_clipper` ?

Reply:
Now it does 😁